opslevel/resource_opslevel_service_dependency.go

```diff
@@ -0,0 +1,108 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opslevel/opslevel-go/v2023"
+)
+
+func resourceServiceDependency() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a service dependency",
+		Create:      wrap(resourceServiceDependencyCreate),
+		Read:        wrap(resourceServiceDependencyRead),
+		Delete:      wrap(resourceServiceDependencyDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"service": {
+				Type:        schema.TypeString,
+				Description: "The ID or alias of the service with the dependency.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"depends_upon": {
+				Type:        schema.TypeString,
+				Description: "The ID or alias of the service that is depended upon.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"note": {
+				Type:        schema.TypeString,
+				Description: "Notes for service dependency.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceServiceDependencyCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	serviceIdentifier := d.Get("service").(string)
+	dependsOn := d.Get("depends_upon").(string)
+
+	input := opslevel.ServiceDependencyCreateInput{
+		Key: opslevel.ServiceDependencyKey{
+			Service:   *opslevel.NewIdentifier(serviceIdentifier),
+			DependsOn: *opslevel.NewIdentifier(dependsOn),
+		},
+		Notes: d.Get("note").(string),
+	}
+	resource, err := client.CreateServiceDependency(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(string(resource.Id))
+
+	return resourceServiceDependencyRead(d, client)
+}
+
+func lookupService(identifier string, client *opslevel.Client) (*opslevel.Service, error) {
```

Comment:
unhappy with this but its because opslevel-go has not been updated to work like this yet - 🤔 maybe i'll fix that